scripts/util/build-doc.js

```diff
@@ -0,0 +1,293 @@
+    console.info(`Getting the commit's tree`);
+    return commit.getTree();
+  }).then(function (tree) {
+    console.info(`Loading the tree into the staging area`);
+    return index.readTree(tree);
+  }).then(function () {
+    console.info(`Staging '${self.options.versionsDirName}' to '${self.options.docsBranchName}'`);
+    return index.addAll(`${self.options.versionsDirName}`, NodeGit.Index.ADD_OPTION.ADD_FORCE);
+  }).then(function () {
+    return repo.getReference(self.options.docsBranchName);
+  }).then(function (reference) {
+    docsBranchResolvedName = reference.name();
+    console.info(`Blew up ${self.options.docsBranchName} into ${docsBranchResolvedName}`);
+    console.info(`Getting the staging area tree id`);
+    return index.writeTree();
+  }).then(function (treeId) {
+    var dateObject = new Date();
+    var time = dateObject.getTime() / 1000;
+    var timeOffset = dateObject.getTimezoneOffset();
+    var author = NodeGit.Signature.create("Auto Docs",
+      "docs@shopify.com", time, timeOffset);
+    var committer = NodeGit.Signature.create("Auto Docs",
+      "docs@shopify.com", time, timeOffset);
+
+    console.info(`Committing changes to '${docsBranchResolvedName}'`);
+    return repo.createCommit(`${docsBranchResolvedName}`, author, committer, 'Docs auto updated during build process', treeId, [docsBranchCommit]); 
+  }).then(function(commitId) {
+    console.info(`New commit created: ${commitId}`);
+    console.info(`Deleting ${self.options.versionsDirName} from the root directory`);
+    recursiveRmdir(path.join('.', self.options.versionsDirName));
+
+    console.info(`Done commiting API docs`);
+    if (callback) {
+      callback(null);
+    }
+  }).catch(function (error) {
+    console.error(`Error encountered while attempting to commit docs to "${self.options.docsBranchName}"`)
+    console.error(error);
+    if (callback) {
+      callback(error);
+    }
+  });
+}
+
+DocBuilder.prototype.build = function (callback) {
+  var self = this;
+
+  self.checkoutDocsBranch(function (error) {
+    if (error) {
+      if (callback) {
+        callback(error);
+      }
+      return;
+    }
+
+    self.checkoutAPISrc(function (error, paths) {
+      if (error) {
+        if (callback) {
+          callback(error);
+        }
+        return;
+      }
+
+      self.generateAPIDocs(JSON.parse(JSON.stringify(paths)), function () {
+
+        if (self.options.rmSrcDir) {
+          console.info('\nRemoving source directories');
+          paths.forEach(function (item) {
+            var srcPath = path.join(item, self.options.srcDirName);
+            console.info(`Removing ${srcPath}`);
+            recursiveRmdir(srcPath);
+          })
+        }
+
+        if (self.options.commitDocs) {
+          self.commitAPIDocs(callback)
+        } else if (callback) {
+          callback();
+        }
+      });
+    });
+  });
+}
+
+module.exports = function (options) {
+  return new DocBuilder(options);
+};
```

Comment:
turn on new line at EOF in your editor.

Also: There is a lot of coupled promise chaining here, which could be broken by a catch statement returning a value. If your then relies on the result of a previous promise, it should be chained off of that promise. There's also a lot of code that lives in every one of these scopes, which should probably be wrapped into functions.